frontend/scripts/metrica.js

```diff
@@ -0,0 +1,231 @@
+(() => {
+    let state = {
+        host: 'http://stage.metrica.fun/api/v1',
+        routes: {
+            create_visitor: '/visitors',
+            create_visit: '/visits'
+        }
+    };
+
+    const getTrackingId = () => {
+        let scripts = document.getElementsByTagName('script');
+        let index = scripts.length - 1;
+        let myScript = scripts[index];
+
+        return (myScript.src.split('tracking_id' + '=')[1] || '').split('&')[0]
+    };
+
+    const createVisitor = (tracking_number) => {
+        return fetch(state.host + state.routes.create_visitor, {
+            method: 'POST',
+            crossDomain: true,
+            headers: {
+                'Content-Type': 'application/json',
+                'x-website': tracking_number
+            },
+        })
+            .then((response) => {
+                return response.json();
+            })
+            .then((result) => {
+                console.log(result.data.token);
+                setToken(result.data.token);
+            });
+    };
+
+    const createVisit = () => {
+        return fetch(state.host + state.routes.create_visit, {
+            method: 'POST',
+            crossDomain: true,
+            headers: {
+                'Content-Type': 'application/json',
+                'x-visitor': 'Bearer ' + getToken()
+            },
+            body: JSON.stringify(getVisit())
+        });
+    };
+
+    const getUserDevice = () => {
+        let device,
+            find,
+            userAgent;
+        device = {};
+
+        window.device = device;
```

Comment:
for global variables use our initial object, like window._metricaTracking, that should be initiated at the same time the script is being started. Actually, the user integrated tracking should initiate that object.